### DISCUSSION TITLE
OSMAnd Pro Cloud Verify Mail is from an IP Address which sends spam

### DISCUSSION THREAD

**Discussion by tabacha** [osmandapp/OsmAnd#12420] [General]
Hi,
I have just subsribed to OSMAnd Pro / Cloud. But I have problems with the activation code. 

I could fix this, because I`m Sysadmin of my own mailserver (perhabs osmand should also do this instead of use bad services). Other user could not fix this kind of error.

Here is the Logfile of my Mailserver:

```
Aug  9 06:57:02 gandalf postfix/smtpd[12644]: NOQUEUE: reject: RCPT from o2.3nn.shared.sendgrid.net[167.89.100.130]: 554 5.7.1 Service unavailable; Client host [167.89.100.130] blocked using ix.dnsbl.manitu.net; Your e-mail service was detected by spam.over.port25.me (NiX Spam) as spamming at Sun, 08 Aug 2021 23:48:56 +0200. Your admin should visit http://www.dnsbl.manitu.net/lookup.php?value=167.89.100.130; from=<bounces+5518261-33f6-osmand2021=sven.anders.hamburg@sendgrid.net> to=<xxxxx@zzz.xxxxx.yyyy> proto=ESMTP helo=<o2.3nn.shared.sendgrid.net>
```  
I attach a screenshot of the wepage http://www.dnsbl.manitu.net/lookup.php?value=167.89.100.130 to this post.

Best Regards

Sven
![image](https://user-images.githubusercontent.com/1101544/128662960-7a2f425b-a28e-4074-8ee8-b2ba2c90ecaa.png)

**Comment by vshcherb** [comment]
Unfortunately we don't have any other email provider, other options were considered to be quite expensive and also there is no guarantee that we will not have same problem. We planned to have own email server but we gave up cause gmail marked our emails as spam

**Reply by tabacha** [reply]
How Expensive? I think you will loose more money, if people are not able to verify a E-Mail for OSMAnd Pro.

**Reply by vshcherb** [reply]
Well definitely, thank you for pointing out and here we will be able to collect feedback how often does this issue occur if we see it regularly and not fixed then we will investigate to switch to another API cause that's not only the budget question but a switching API as well
